cip-tbd/cip-tbd.md

```diff
@@ -0,0 +1,503 @@
+
+##### connect
+
+Returns a `ConnectResult` with `isConnected = true` if authentication between the user client and the dApp API server has been successfully established; otherwise, returns a `ConnectResult` with `isConnected = false`.
+Additionally, the response **MAY** include an `isNetworkConnected` field (optionally accompanied by an error reason) indicating whether the dApp API server is currently connected to a Canton network.
+- Network connection is considered to be established if an unauthenticated ledger endpoint, such as `/v2/version`, can be accessed
+
+If the user client does not have an active session with the server, the user login flow **MUST** be executed before a response is returned.
+
+```typescript
+// ConnectResult
+{
+  isConnected: boolean,
+  reason?: string
+  isNetworkConnected?: boolean,
+  networkReason?: string
+}
+```
+
+##### isConnected
+
+Same as `connect`, but does not initialize the login.
+
+##### disconnect
+
+Closes the connection between the user client and the dApp API server. As a result, the user session, involving the access token, is being invalidated on the server. In addition, the `statusChanged` event **MAY** be emitted as an asynchronous confirmation sent back to the dApp.
+
+##### status
+
+Returns the connectivity status as per `connect` and additional information regarding the connection and the dApp API server properties. The property `kernel` holds a unique identifier of the server (e.g., the Wallet component) and indicates the deployment type in the `clientType` field. In addition, the optional fields `network` and `session` **MAY** carry information about the user's session and the network to which the connection is established. However, if returned, the `networkId` **SHOULD** be [CAIP-2](https://chainagnostic.org/CAIPs/caip-2) compliant, e.g., `canton:da-mainnet`.
+
+```typescript
+// StatusEvent
+{
+connection: ConnectResult,
+provider: Provider,
+network?: Network,
+session?: Session
+}
+
+// Provider
+{
+  id: string,
+  version: string, // dApp API version
+  providerType: Enum[browser,desktop,mobile,remote]
+}
+
+// Network
+{
+  networkId: string, // CAIP-2
+  ledgerApi?: string
+  accessToken?: string,
+}
+
+// Session
+{
+  accessToken: string,
+  userId: string
+}
+```
+
+##### getActiveNetwork
+
+Returns the network (same type as seen in `status`) with which the user has established a connection via the dApp server.
+
+##### listAccounts
+
+Return a list of accounts the user has access to. Specifically, an account is a Canton `party` with additional metadata, as well as information about its status and the network where its private data can be accessed. In addition, the user chooses to use a given account as the `primary` account (and the corresponding network) in the context of the current dApp session. 
+
+```typescript
+// Account
+{
+  primary: boolean,
+  partyId: string,
+  status: Enum[initializing, allocated],
+  hint: string,
+  publicKey: string,
```

Comment:
Is there a format specified for the publicKey? Hex? What about the curve chosen?